src/arch/arm/64/machine/fpu.c

```diff
@@ -30,10 +33,10 @@ BOOT_CODE bool_t fpsimd_HWCapTest(void)
 
     /* Check if the hardware has FP and ASIMD support... */
     MRS("id_aa64pfr0_el1", id_aa64pfr0);
-    if (((id_aa64pfr0 >> ID_AA64PFR0_EL1_FP) & MASK(4)) == MASK(4) ||
-        ((id_aa64pfr0 >> ID_AA64PFR0_EL1_ASIMD) & MASK(4)) == MASK(4)) {
-        return false;
+    if (IS_FPU_ENABLED(id_aa64pfr0, ID_AA64PFR0_EL1_FP) ||
+        IS_FPU_ENABLED(id_aa64pfr0, ID_AA64PFR0_EL1_ASIMD)) {
+        return true;
```

Comment:
The logic is changed from returning false when either FP or ASIMD is not supported to only returning false when both are unsupported.

If intentional, perhaps change the commit message to make this explicit and explain why it's okay? The current message does not match the change made, as the old code already checked for 0xF and there is no explicit check for 0x1 added.

The original check seems to assume that either both or none are present, and that assumption might be propagated elsewhere. What will happen now if FP is supported, but ASIMD isn't?

Reply:
I think the new commit makes sense since as I recall, VFP and ASIMD registers are mapped to the same underlying hardware registers. So we need to switch FPU registers when either of them (most of the time, both) is implemented.